plugins/woocommerce/src/Internal/Admin/WcPaySubscriptionsPage.php

```diff
@@ -74,6 +71,80 @@ public function register_subscriptions_page() {
 		}
 	}
 
+	/**
+	 * Returns true if the store is eligible for the WooCommerce subsciptions empty state experiment.
+	 *
+	 * @return bool
+	 */
+	private function is_store_experiment_eligible() {
+		// Ineligible if WooCommerce Payments OR an existing subscriptions plugin is installed.
+		$installed_plugins      = PluginsHelper::get_installed_plugin_slugs();
+		$plugin_ineligible_list = array(
+			'woocommerce-payments',
+			'woocommerce-subscriptions',
+			'subscriptio',
+			'subscriptions-for-woocommerce',
+			'subscriptions-for-woocommerce-pro',
+			'sumosubscriptions',
+			'yith-woocommerce-subscription',
+			'xa-woocommerce-subscriptions',
+		);
+		foreach ( $plugin_ineligible_list as $plugin_slug ) {
+			if ( in_array( $plugin_slug, $installed_plugins, true ) ) {
+				return false;
+			}
+		}
+
+		// Ineligible if store address is not compatible with WCPay Subscriptions (US).
+		$store_base_location = wc_get_base_location();
+		if ( empty( $store_base_location['country'] ) || 'US' !== $store_base_location['country'] ) {
+			return false;
+		}
+
+		// Ineligible if store has not been active for at least 6 months.
+		if ( ! WCAdminHelper::is_wc_admin_active_in_date_range( 'month-6+' ) ) {
+			return false;
+		}
+
+		// Ineligible if store has not had any sales in the last 30 days.
+		if ( ! $this->get_store_recent_sales_eligibility() ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns true if the store has an order completed within the last 30 days.
+	 *
+	 * @return bool
+	 */
+	private function get_store_recent_sales_eligibility() {
+		$transient_key = 'woocommerce-wcpay-subscriptions_recent_sales_eligibility';
+
+		// Load from cache.
+		$is_eligible = get_transient( $transient_key );
+
+		// Valid cache found.
+		if ( false !== $is_eligible ) {
+			return $is_eligible;
+		}
+
+		// Get a single order that was marked as completed within the last 30 days.
+		$orders = wc_get_orders(
+			array(
+				'date_paid' => '>' . strtotime( '-30 days' ),
+				'limit'     => 1,
+				'return'    => 'ids',
+			)
+		);
+
+		$is_eligible = count( $orders ) >= 1;
+		set_transient( $transient_key, $is_eligible, DAY_IN_SECONDS );
```

Comment:
The docs recommend against storing a boolean value in transients, we might want to store a string instead, e.g. `yes` and `no`.

>Returned value is the value of transient. If the transient does not exist, does not have a value, or has expired, then get_transient will return `false`. This should be checked using the identity operator ( `===` ) instead of the normal equality operator, because an integer value of zero (or other “empty” data) could be the data you’re wanting to store. Because of this “false” value, transients should not be used to hold plain boolean values. Put them into an array or convert them to integers instead.

https://developer.wordpress.org/reference/functions/get_transient/#more-information

Reply:
If there's a more readable or robust way to store boolean values in transients, we should go with that. I know that woo tends to use `yes` and `no` for options, maybe that's safer and clearer and more consistent.

Reply:
Thanks @haszari, I've changed the stored value from `eligible`/`not_eligible` to `yes`/`no` in 1aa88a9

Reply:
WC have helpers to convert bools and strings for storage and for retrieval. We could use those to avoid having to wrangle the storage values. 

bool -> `wc_bool_to_string()` -> 'yes' / 'no'
'yes' / 'no' / 'true' / 'false' -> `wc_string_to_bool()` -> bool

Reply:
@james-allan thanks! Changed to use these WC formatting functions in ffc64e9